### PR TITLE
Prepare v0.101.0 release

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:** 
Prepare v0.101.0 release

**Link to tracking Issue(s):** 
Related to https://github.com/open-telemetry/opentelemetry-operator/issues/2983

### CODE REVIEW

**Review by iblancasa — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,30 @@
...
+  A new featuregate `operator.golang.flags` is added. This featuregate will allow the operator to automatically
+  set GOMAXPROCS and GOMEMLIMIT equal to the CPU and Memory limit provided respectively for the pod. 
+  
+
+### Components
+
+* [OpenTelemetry Collector - v0.101.0](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.101.0)
+* [OpenTelemetry Contrib - v0.101.0](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.101.0)
+* [Java auto-instrumentation - v1.32.1](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v1.32.1)
+* [.NET auto-instrumentation - v1.2.0](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/releases/tag/{AUTO_INSTRUMENTATION_DOTNET_VERSION})
```
Comment (jaronoff97) [inline CHANGELOG.md:22]: instrumentation version didn't get autofilled correctly

File: README.md [hunk 2]
```diff
@@ -730,6 +730,7 @@ The OpenTelemetry Operator _might_ work on versions outside of the given range,
 
 | OpenTelemetry Operator | Kubernetes     | Cert-Manager |
 |------------------------| -------------- | ------------ |
+| v0.101.0               | v1.23 to v1.29 | v1           |
```
Comment (jaronoff97) [inline README.md:733]: i believe this should be bumped to 1.30 now that #2975 is merged?

File: CHANGELOG.md (hunk 1 above)
Comment (TylerHelmuth) [reply CHANGELOG.md]: Fixed (several places)

File: README.md (hunk 2 above)
Comment (TylerHelmuth) [reply README.md]: Updated

**Review by jaronoff97 — APPROVED** [review]

**Review by swiatekm-sumo — APPROVED** [review]

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**